torch/_inductor/config.py

```diff
@@ -27,23 +27,30 @@ def fx_graph_remote_cache_default():
 # Whether to enable printing the source code for each future
 verbose_progress = False
 
+# Force disabled all inductor level caching -- This will override any other caching flag
+force_disable_caches = os.environ.get("TORCHINDUCTOR_FORCE_DISABLE_CACHES") == "1"
+_enable_caches = not force_disable_caches
+
 # use fx aot graph codegen cache
-fx_graph_cache = os.environ.get("TORCHINDUCTOR_FX_GRAPH_CACHE") == "1"
+fx_graph_cache = (
```

Comment:
i dont think this is enough actually. This is decided at import time, so if you later set config.force_disable_caches = True, it will not have an impact on fx_graph_cache

Reply:
You want to support

```
import torch

config.force_disable_caches = False

torch.compile(...)
```

use case

Reply:
I actually kinda didn't want to support that, but ok, I guess I was being lazy. In that case, the way these settings are structured I don't know how to do it except update every point in code that checks for some cache being enabled. I wish I knew a better way.

Reply:
Oh, I remembered (partly) why I was being lazy. After you import torch, the process pool has already started so the `config.force_disable_caches = False` is too late :/

Reply:
> update every point in code that checks for some cache being enabled

This is kinda what i did for most of them, which you undid in this PR.

> After you import torch, the process pool has already started so the config.force_disable_caches = False is too late

is there no way to broadcast to the thread pool?

Reply:
Yeah, I'd convinced myself that it was fine to expose this feature only via env var; I guess because it makes the problem much easier. It would also be much easier if Triton just allowed disabling the cache.

> is there no way to broadcast to the thread pool?

They're actually sub-processes. I think it would have to be a custom protocol. 

Still thinking...

Reply:
> It would also be much easier if Triton just allowed disabling the cache.

We can add this, happy to write a PR to Triton for this. cc: @bertmaher @htyu

Reply:
Actually I think I can work around it using the existing approach of creating temp dirs. It would be a bit cleaner if we get the ability to turn off the Triton cache in the future.

Reply:
Triton provides a way to bypass caching by always recompiling the kernel via `TRITON_ALWAYS_COMPILE=1`. Would that be helpful?

Reply:
@htyu, aha, yes it would. Thanks!